core/adapters/random.go

```diff
@@ -108,11 +164,16 @@ func getKey(ra *Random, input models.RunInput) (*vrfkey.PublicKey, error) {
 }
 
 // extractHex returns the bytes corresponding to the string input at the key
-// field, or an error.
+// field, or an error if the string cannot be interpreted as a hex string or
+// represents more than 32 bytes
 func extractHex(input models.RunInput, key string) ([]byte, error) {
 	rawValue := input.Data().Get(key)
 	if rawValue.Type != gjson.String {
 		return nil, fmt.Errorf("%s %#+v is not a hex string", key, rawValue)
 	}
+	if len(rawValue.String()) > 66 {
```

Comment:
I can haz comment explaining this magic number? Or even better, move it into an appropriately named constant.